modules/auxiliary/server/capture/ldap.rb

```diff
@@ -0,0 +1,106 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::LDAP::Server
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Authentication Capture: LDAP',
+        'Description' => %q{
+          This module mocks an LDAP service to capture authentication
+          information of a client trying to authenticate against an LDAP service
+        },
+        'Author' => 'JustAnda7',
+        'License' => MSF_LICENSE,
+        'Action' => [
+          [ 'Capture', { 'Description' => 'Run an LDAP capture server' } ]
+        ],
+        'PassiveActions' => [ 'Capture' ],
+        'DefaultActions' => 'Capture',
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptAddress.new('SRVHOST', [ true, 'The ip address to listen on.', '0.0.0.0' ]),
+        OptPort.new('SRVPORT', [ true, 'The port to listen on.', '389' ]),
+        OptString.new('CHALLENGE', [ true, 'The 8 byte challenge', Rex::Text.rand_text_alphanumeric(16) ])
+      ]
+    )
+
+    deregister_options('LDIF_FILE')
+
+    register_advanced_options(
+      [
+        OptString.new('Domain', [ false, 'The default domain to use for NTLM authentication', 'DOMAIN']),
+        OptString.new('Server', [ false, 'The default server to use for NTLM authentication', 'SERVER']),
+        OptString.new('DnsName', [ false, 'The default DNS server name to use for NTLM authentication', 'SERVER']),
+        OptString.new('DnsDomain', [ false, 'The default DNS domain name to use for NTLM authentication', 'example.com']),
```

Comment:
Should these options be required?

Reply:
No there are defaults provided and these would not be used at all during a simple auth attempt.